include/deal.II/grid/manifold_lib.h

```diff
@@ -461,6 +461,83 @@ class CylindricalManifold : public ChartManifold<dim, spacedim, 3>
   double tolerance;
 };
 
+/**
+ * Elliptical manifold description derived from ChartManifold.
+ * More informations on the elliptical coordinate system can be found
+ * at <a
+ * href="https://en.wikipedia.org/wiki/Elliptic_coordinate_system">Wikipedia
+ * </a>.
+ *
+ * The constructor of this class will throw an exception if both dim and
+ * spacedim are different from two.
+ *
+ * This manifold can be used to produce hyper_shells with elliptical curvature.
+ * As an example, the test elliptical_manifold_01 produces the following
+ * triangulation:
+ * @image html elliptical_hyper_shell.png
+ *
+ * @ingroup manifold
+ *
+ * @author Stefano Dominici, 2018
+ */
+template <int dim, int spacedim = dim>
+class EllipticalManifold : public ChartManifold<dim, spacedim, spacedim>
+{
+public:
+  /**
+   * Constructor that takes the center of the manifold system, the direction of
+   * the major axis, and the eccentricity parameter.
+   * The default major axis is the <tt>x</tt>-axis. The manifold is rotated in
+   * order to align the major axis to the direction specified in input.
+   * @param center Center of the manifold.
+   * @param major_axis_direction Direction of the major axis of the
+   * manifold.
+   * @param c_parameter Parameter controlling the eccentricity of the
+   * manifold.
+   *
+   */
+  EllipticalManifold(const Point<spacedim> &    center,
+                     const Tensor<1, spacedim> &major_axis_direction,
+                     const double               c_parameter);
+  virtual ~EllipticalManifold() = default;
```

Comment:
Correct. Declaring a destructor as `virtual ~C() = default;` has the only effect of making sure the destructor is indeed `virtual`. Other than that, this construct is identical to letting the compiler generate the destructor itself.

But since the class is derived from a base class that *already* marks the destructor as `virtual`, the destructor of the current class is automatically `virtual`, whether defined as above or generated automatically by the compiler.